go.mod

```diff
@@ -2,6 +2,8 @@ module github.com/spacemeshos/go-spacemesh
 
 go 1.19
 
+replace github.com/spacemeshos/api/release/go v1.18.0 => github.com/spacemeshos/api/release/go v1.18.1-0.20230811144450-baf8688a08b2
```

Comment:
Please do not use `replace`. This is not its intended use, you can directly update the dependency instead.

`replace` is intended to be able to forcefully overwrite transient dependencies when e.g. a critical vulnerability needs to be fixed and the dependency using it is slow to update. `replace` statements are not applied to dependencies of `github.com/spacemeshos/go-spacemesh` and ignored by modules importing `github.com/spacemeshos/go-spacemesh`.

So any module importing `github.com/spacemeshos/go-spacemesh`needs to copy all `replace` statements listed here or they might not even compile, see also https://go.dev/ref/mod#go-mod-file-replace

Reply:
Sorry I meant to leave a note on the PR, the replace is temporary. I don't know what is the exact process we use when updating the API package but I assume we need to release it and update this PR to point to the release before it can be merged, at which point I planned to remove the replace.

Reply:
You can directly go get into the `go-spacemesh` branch. Just make sure to merge & release the API changes before you merge and release the `go-spacemesh` changes 🙂

Reply:
Another thing is if one forgets the `replace` statement it will be there until someone removes it, if one updates the dependency directly and forgets to update again when the API repo releases a new version, @dependabot will remind us 🙂

Reply:
Noted, thanks!